src/moditect/module-info.java

```diff
@@ -0,0 +1,11 @@
+module com.fasterxml.jackson.core {
+	exports com.fasterxml.jackson.core;
+
+	exports com.fasterxml.jackson.core.type to com.fasterxml.jackson.databind, com.jwebmp.core, com.fasterxml.jackson.datatype.jsr310;
```

Comment:
Should be open to all: `TypeReference` (for example) is very widely used and essential part of API.

Reply:
hmm, it also seems to have my personal libraries listed there xD 

Going top down - 
Question 1)
- For internal API's that the library references, in this case, parent/bom/core/databind/annotations etc, these exports list should be specific to the internal references. that jwebmp.com one should definitely not be there. In terms of allowing modules to expose, you would normally never "open" a module, unless absolutely necessary, and definitely not expose all packages (same effect). Yes it seems refactoring would need to be done into a "impl" package of sorts that would be exposed. 
The idea is one exported package per module (I haven't managed to get this right yet anywhere). This is to minimize a) open api classes, and b) restrict public class viewing to only the internal API's that need it, and c) enforce proper object orientation encapsulation, package level.

Question 2) 
The java compiler can build previous class versions, yes the compiler and maven build will run in JDK 8 (hence why I said that Travis build will break on JDK 7), and you set the compiler to read sources and generate targets at a 1.6 JDK level. The pull request does this already :)

Agreed a list of packages to be exported would need to be compiled, but in my experience, it's better to work off a negative list, rather than a positive one. Just now you have a random someone using a class in your utils package which is a massive no-no....

OK, hope I answered most of your JPMS questions! I don't think the packages need to be redefined at all, or any refactoring has to take place. In terms of opening minor packages for known implementations, this can of course be done, when the packages get large with a lot of classes, serious consideration needs to take place with regards to exporting them. Don't break the module road map in the new JVM ;)

Reply:
Ok. I think I was confused by removal of Java 6 source/target, as I did not see the replacement.
Build on Java 8 is fine, given that I don't think I can do successful Maven Central deployment with pre-Java-8 tools any more (for some ssl certificate thing or another).

As to opening up, I think we agree on idea. I was just looking at proposed changes and they seemed quite specific, but probably they were generated by a tool, for minimal exposure?

Reply:
Ha! I'll take that as a compliment!!

I've gone through a few routes on this journey of pure module pathing, and I can honestly say it definitely is easier going from nothing up :)

On one of the harder ones (Guice), they attach the classes of asm and cglib in, and you need to expose that as well, but you can't during a regular build, so there are funnies like these that are resolvable, but it feels like we're all starting a new journey xD

When looking at the faster xml module pathing, I came up with the following - 
The really kak one is the java.desktop, and I think perhaps that a datatypes module for sql to keep it perfectly clean is probably a good idea, and the one class that is using it (i can't remember which it is now but removing it will highlight which one)

![image](https://user-images.githubusercontent.com/5367513/49542905-7179c000-f8df-11e8-8a1f-4318b430da00.png)

The important thing here is that no paths are duplicated, except for the cores java.logging and java.xml which is perfectly fine 👍 
This is bit different to the current PR (I did a bit of work on it once everything was named) - I'll try PR this all in tonight.

When looking at the "service-modules" the structure stays in tact. This is the benefit of named modules and i've confirmed jlink build compatible.

![image](https://user-images.githubusercontent.com/5367513/49543079-e64cfa00-f8df-11e8-8078-c76f19459925.png)
![image](https://user-images.githubusercontent.com/5367513/49543102-f533ac80-f8df-11e8-8815-9fb2be837fbd.png)

Reply:
hmm see still a few more duplicate dependents xD